ui/elements/Modal/Modal.tsx

```diff
@@ -120,81 +116,89 @@ const Modal = forwardRef<HTMLDivElement, ModalProps>((props, ref) => {
                     animationDuration={0.3}
                     shouldFocusOnFirstElement={shouldFocusOnFirstElement}
                 >
-                    <StyledModal
-                        onClick={(e: React.MouseEvent<HTMLDivElement>) =>
-                            e.stopPropagation()
-                        }
-                        ref={modalRef}
-                        width={width}
-                        height={height}
-                        minHeight={minHeight}
-                        minWidth={minWidth}
-                        maxHeight={maxHeight}
-                        maxWidth={maxWidth}
-                        mobileBottomFullWidth={mobileBottomFullWidth}
-                        modalPadding={modalPadding}
-                        $border={border}
-                        aria-labelledby={`modal-title-${id}`}
-                        aria-describedby={`modal-description-${id}`}
-                        aria-modal="true"
-                        role="dialog"
-                        data-testid="testid-modal"
-                        {...motionProps}
+                    <FocusTrap
+                        active={isOpen}
+                        focusTrapOptions={{
+                            initialFocus: shouldFocusOnFirstElement ? undefined : false,
+                            returnFocusOnDeactivate: true,
+                        }}
```

Comment:
Lets make this feature and its initialFocus and returnFocusOnDeactivate properties configurable via the modal's props

Reply:
a suggestion would be to create a separate type for FocusTrap props that we need to be configurable (in `common/types.ts`), and make ModalProp type extend the FocusTrapProp type